### PR TITLE
System Cleanup 

### PR DESCRIPTION
**Pull request by mashrukk**
## System Cleanup Script

This is a comprehensive system cleanup script that performs various cleanup tasks.

#### Key Features:
- Removes unused packages
- Common cleanup tasks (journalctl, temporary files, log truncation)
- Optional user cache and trash cleanup

This script provides a unified approach to system cleanup across different Linux distributions, improving system performance and freeing up disk space.

## Type of Change
- [x] New feature

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tabs/system-setup/system-cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#!/bin/sh -e
+
+. ../common-script.sh
+
+cleanup_system() {
+    printf "%b\n" "${YELLOW}Performing system cleanup...${RC}"
+    case $PACKAGER in
+        apt-get|nala)
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:8]:
```suggestion
        apt|nala)
```

File: tabs/system-setup/system-cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            $ESCALATION_TOOL "$PACKAGER" clean
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:9]:
```suggestion
            "$ESCALATION_TOOL" "$PACKAGER" clean
```
Quote $escalation_tool for every one

File: tabs/system-setup/system-cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#!/bin/sh -e
+
+. ../common-script.sh
+
+cleanup_system() {
+    printf "%b\n" "${YELLOW}Performing system cleanup...${RC}"
+    case $PACKAGER in
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:7]:
```suggestion
    case "$PACKAGER" in
```

File: tabs/system-setup/system-cleanup.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            $ESCALATION_TOOL "$PACKAGER" remove "$(dnf repoquery --extras --exclude=kernel,kernel-\*)" -y
+            ;;
+        pacman)
+            $ESCALATION_TOOL "$PACKAGER" -Sc --noconfirm
+            $ESCALATION_TOOL "$PACKAGER" -Rns "$(pacman -Qtdq)" --noconfirm
+            if command_exists paru >/dev/null 2>&1; then
+                paru -c
+            elif command_exists yay >/dev/null 2>&1; then
+                yay -c
+            fi
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:32]:
```bash
            if command_exists paru >/dev/null 2>&1; then
                paru -c
            elif command_exists yay >/dev/null 2>&1; then
                yay -c
            fi
```
This is redundant

File: tabs/system-setup/system-cleanup.sh [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            ;;
+        *)
+            printf "%b\n" "${RED}Unsupported package manager.${RC}"
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:35]:
```suggestion
            printf "%b\n" "${RED}Unsupported package manager: $PACKAGER${RC}"
```

File: tabs/system-setup/system-cleanup.sh [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+common_cleanup() {
+    $ESCALATION_TOOL rm -rf /var/tmp/*
+    $ESCALATION_TOOL rm -rf /tmp/*
+    $ESCALATION_TOOL find /var/log -type f -name "*.log" -exec truncate -s 0 {} \;
+    $ESCALATION_TOOL journalctl --vacuum-time=3d
+}
+
+clean_data() {
+    printf "%b" "${YELLOW}Clean up old cache files and empty the trash? (y/N) ${RC}"
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:51]:
```suggestion
    printf "%b\n" "${YELLOW}Clean up old cache files and empty the trash? (y/N): ${RC}"
```

File: tabs/system-setup/system-cleanup.sh [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    read -r clean_response
+    case $clean_response in
+        [Yy]*)
+            printf "%b\n" "${YELLOW}Cleaning up old cache files and emptying trash...${RC}"
+            find ~/.cache/ -type f -atime +5 -delete
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:56]:
```suggestion
            find "$HOME/.cache/" -type f -atime +5 -delete
```

File: tabs/system-setup/system-cleanup.sh [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+            find ~/.local/share/Trash -mindepth 1 -delete
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:57]:
```suggestion
            find "$HOME/.local/share/Trash" -mindepth 1 -delete
```

File: tabs/system-setup/system-cleanup.sh [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+            ;;
+        zypper)
+            $ESCALATION_TOOL "$PACKAGER" clean -a
+            $ESCALATION_TOOL "$PACKAGER" tidy
+            $ESCALATION_TOOL "$PACKAGER" cc -a
+            ;;
+        dnf)
+            $ESCALATION_TOOL "$PACKAGER" clean all
+            $ESCALATION_TOOL "$PACKAGER" autoremove -y
+            $ESCALATION_TOOL "$PACKAGER" remove "$(dnf repoquery --extras --exclude=kernel,kernel-\*)" -y
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:23]:
```suggestion
```

File: tabs/system-setup/system-cleanup.sh [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+            ;;
+    esac
+
+    printf "%b\n" "${GREEN}System cleanup completed.${RC}"
+}
+
+common_cleanup() {
+    $ESCALATION_TOOL rm -rf /var/tmp/*
+    $ESCALATION_TOOL rm -rf /tmp/*
+    $ESCALATION_TOOL find /var/log -type f -name "*.log" -exec truncate -s 0 {} \;
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:40]: All of these $ESCALATION_TOOL need quotes..

EX: ``"$ESCALATION_TOOL"`` not ``$ESCALATION_TOOL``

This needs to be done for every escalation tool

Comment (mashrukk) [reply tabs/system-setup/system-cleanup.sh:40]: You mean every single one?

File: tabs/system-setup/system-cleanup.sh [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+            printf "%b\n" "${RED}Unsupported package manager: $PACKAGER${RC}"
+            return 1
+            ;;
+    esac
+
+    printf "%b\n" "${GREEN}System cleanup completed.${RC}"
+}
+
+common_cleanup() {
+    "$ESCALATION_TOOL" rm -rf /var/tmp/*
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:38]:
```suggestion
```
This is not good, systemd uses this dir for storing temp files. Remove this.

Comment (nnyyxxxx) [reply tabs/system-setup/system-cleanup.sh:38]: Or make it only delete files that are a week old++

File: tabs/system-setup/system-cleanup.sh [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    "$ESCALATION_TOOL" rm -rf /tmp/*
```
Comment (nnyyxxxx) [inline tabs/system-setup/system-cleanup.sh:39]:
```suggestion
```
Same with this, remove it or make it only delete files a week old++

**Pull request closed without merging**